sklearn/gp_search.py

```diff
@@ -0,0 +1,421 @@
+
+
+class GPSearchCV(BaseSearchCV):
+    """
+    Parameters
+    ----------
+
+    estimator : 1) sklearn estimator or 2) callable
+        1 : object type that implements the "fit" and "predict" methods,
+        as a classifier or a pipeline
+        2 : a function that computes the output given a dictionnary of
+        parameters. The returned value should be a list of one or more
+        floats if score_format == 'cv', and a float if score_format ==
+        'avg'
+
+    parameters : dict, parameter space on which to optimize the estimator
+        The keys of the dictionnary should be the names of the parameters,
+        and the values should be lists of length 2; the first element being
+        the type of the parameter ('int', 'float' or 'cat' [for categorical]),
+        and the second element being a list of either the bounds between which
+        to search (for 'int' and 'float') or the values the parameter can take
+        (for 'cat')
+        Example : parameters = {'kernel' :  ['cat', ['rbf', 'poly']],
+                                 'd' : ['int', [1,3]],
+                                 'C' : ['float',[1,10])}
+
+    scoring : string, callable or None, optional
+        A string (see sklearn's model evaluation documentation) or
+        a scorer callable object / function with signature
+        ``scorer(estimator, X, y)``.
+        Default is None.
+
+    fit_params : dict, optional
+        Parameters to pass to the fit method.
+
+    cv : integer or cross-validation generator, optional
+        Relevant if the estimator is an sklearn object.
+        If an integer is passed, it is the number of folds.
+        Specific cross-validation objects can be passed, see
+        sklearn.cross_validation module for the list of possible objects
+        Default is 5.
+
+    acquisition function : string, optional
+        Function to maximize in order to choose the next parameter to test.
+        - Simple : maximize the predicted output
+        - UCB : maximize the upper confidence bound
+        - EI : maximizes the expected improvement
+        Default is 'UCB'
+
+    n_iter : int
+        Total number of iterations to perform (including n_init and
+        n_final_iter).
+        Default is 100.
+
+    n_init : int, optional
+        Number of random iterations to perform before the smart search.
+        Default is 30.
+
+    n_final_iter : int, optional
+        Number of final iterations, ie. smart iterations but with
+        acquisition_function == 'Simple'
+        Default is 5.
+
+    n_candidates : int, optional
+        Number of random candidates to sample for each GP iterations
+        Default is 500.
+
+    nugget : float, optional
+        The nugget to set for the Gaussian Process.
+        Default is 1.e-10.
+
+
+    Attributes
+    ----------
+
+    best_params_ : dict, the parameter set, from those tested by the
+        method _fit, that maximizes the mean of the cross-validation results.
```

Comment:
`best_score_` and `best_estimator_` should also be provided in my opinion, to have a consistent API with GridSearchCV and RandomizedSearchCV.